dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -826,6 +827,149 @@ ShowStatusPropertyDialog(
+    UINT uEnableDisableCmd = IDM_NETICON_ENABLE;
+    UINT uEnableDisableString = IDS_NET_ACTIVATE;
+
+    if (SUCCEEDED(hr) && pProps)
+    {
+        if (pProps->Status == NCS_HARDWARE_DISABLED ||
+            pProps->Status == NCS_MEDIA_DISCONNECTED ||
+            pProps->Status == NCS_DISCONNECTED)
+        {
+            uEnableDisableFlags = MF_STRING | MFS_DEFAULT;
+            uEnableDisableCmd = IDM_NETICON_ENABLE;
+            uEnableDisableString = IDS_NET_ACTIVATE;
+            uPropertiesFlags = MF_STRING;
+        }
+        else
+        {
+            uEnableDisableFlags = MF_STRING;
+            uEnableDisableCmd = IDM_NETICON_DISABLE;
+            uEnableDisableString = IDS_NET_DEACTIVATE;
+        }
+
+        if (pProps->Status == NCS_CONNECTED)
+        {
+            uStatusFlags = MF_STRING;
+            uRepairFlags = MF_STRING;
+        }
+        else if (pProps->Status == NCS_CONNECTING)
+        {
+            uStatusFlags = MF_STRING;
+            uRepairFlags = MF_STRING | MF_GRAYED;
+        }
+        else
+        {
+            uStatusFlags = MF_STRING | MF_GRAYED;
+            uRepairFlags = MF_STRING | MF_GRAYED;
+        }
+
+        NcFreeNetconProperties(pProps);
+        pProps = NULL;
+    }
+    else
+    {
+        uEnableDisableFlags = MF_STRING | MF_GRAYED;
+        uStatusFlags = MF_STRING | MF_GRAYED;
+        uRepairFlags = MF_STRING | MF_GRAYED;
+        uPropertiesFlags = MF_STRING | MF_GRAYED;
+    }
+
+
+    WCHAR szMenuItem[128];
+
+    if (LoadStringW(netshell_hInstance, uEnableDisableString, szMenuItem, _countof(szMenuItem)))
+    {
+        AppendMenuW(hMenu, uEnableDisableFlags, uEnableDisableCmd, szMenuItem);
+    }
+
+    if (LoadStringW(netshell_hInstance, IDS_NET_STATUS, szMenuItem, _countof(szMenuItem)))
+    {
+        AppendMenuW(hMenu, uStatusFlags, IDM_NETICON_STATUS, szMenuItem);
+    }
+
+    if (LoadStringW(netshell_hInstance, IDS_NET_REPAIR, szMenuItem, _countof(szMenuItem)))
+    {
+        AppendMenuW(hMenu, uRepairFlags, IDM_NETICON_REPAIR, szMenuItem);
+    }
+
+    AppendMenuW(hMenu, MF_SEPARATOR, 0, NULL);
+
+    if (LoadStringW(netshell_hInstance, IDS_NET_OPEN_CONNECTIONS, szMenuItem, _countof(szMenuItem)))
+    {
+        AppendMenuW(hMenu, MF_STRING, IDM_NETICON_OPEN_CONNECTIONS, szMenuItem);
+    }
+
+    if (LoadStringW(netshell_hInstance, IDS_NET_PROPERTIES, szMenuItem, _countof(szMenuItem)))
+    {
+        UINT finalPropertiesFlags = uPropertiesFlags;
+        if (!(uEnableDisableFlags & MFS_DEFAULT)) {
+            finalPropertiesFlags |= MFS_DEFAULT;
+        }
+        AppendMenuW(hMenu, finalPropertiesFlags, IDM_NETICON_PROPERTIES, szMenuItem);
+    }
+
+    TrackPopupMenuEx(hMenu, TPM_LEFTALIGN | TPM_BOTTOMALIGN | TPM_RIGHTBUTTON, pt.x, pt.y, hwndOwner, NULL);
+
+    PostMessage(hwndOwner, WM_NULL, 0, 0);
+
+    DestroyMenu(hMenu);
+}
+
+VOID LaunchNetworkConnectionsFolder(HWND hwndOwner)
+{
+    WCHAR szCommand[] = L"control.exe netconnections";
```

Comment:
You could probably simplify it like that:
```suggestion
    WCHAR szCommand[] = L"control netconnections";
```